Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISCreateTransmission.py

```diff
@@ -0,0 +1,179 @@
+        len_validator = StringArrayLengthValidator()
+        len_validator.setLengthMin(1)
+        mandatory_runs.add(len_validator)
+        self.declareProperty(
+            StringArrayProperty(Prop.INPUT_RUNS, values=[], validator=mandatory_runs),
+            doc="A list of input run numbers. Multiple runs will be summed after loading",
+        )
+
+        self.copyProperties(self._TRANS_WS_ALG, [Prop.TRANS_ROI, Prop.I0_MON_IDX, Prop.MON_WAV_MIN, Prop.MON_WAV_MAX])
+
+        self.declareProperty(
+            MatrixWorkspaceProperty(Prop.FLOOD_WS, "", direction=Direction.Input, optional=PropertyMode.Optional),
+            doc="The workspace to be used for the flood correction. If this property is not set then no flood correction is performed.",
+        )
+
+        self.declareProperty(
+            Prop.BACK_SUB_ROI,
+            "",
+            doc=f"The set of workspace indices to be passed to the ProcessingInstructions property of the {self._BACK_SUB_ALG} algorithm."
+            " If this property is not set then no background subtraction is performed.",
+        )
+
+        self.declareProperty(
+            WorkspaceProperty(Prop.OUTPUT_WS, "", direction=Direction.Output),
+            doc="The output transmission workspace",
+        )
+
+    def PyExec(self):
+        ws = self._load_and_sum_runs(self.getProperty(Prop.INPUT_RUNS).value, self.getPropertyValue(Prop.OUTPUT_WS))
+
+        ws = self._apply_flood_correction(ws)
+        ws = self._apply_background_subtraction(ws)
+        ws = self._create_transmission_ws(ws)
+
+        self.setProperty(Prop.OUTPUT_WS, ws)
+
+    def _load_and_sum_runs(self, runs: list[str], output_name: str):
+        """Load and sum the input runs"""
+        self.log().information("Loading and summing the run files")
+        alg = self.createChildAlgorithm(self._LOAD_ALG, Filename="+".join(runs), LoaderName="LoadNexus", OutputWorkspace=output_name)
+        alg.setRethrows(True)
+        alg.setProperty("MergeRunsOptions", {"FailBehaviour": "Stop"})
+        alg.execute()
+        return alg.getProperty("OutputWorkspace").value
+
+    def _apply_flood_correction(self, workspace):
+        """If a flood workspace has been provided then apply a flood correction"""
+        flood_ws = self.getProperty(Prop.FLOOD_WS).value
+        if flood_ws is None:
+            return workspace
+
+        self.log().information("Performing flood correction")
+        args = {"FloodWorkspace": flood_ws}
+        corrected_ws = self._run_algorithm(workspace, self._FLOOD_ALG, "InputWorkspace", args)
+
+        # The flood correction performs a divide, which can result in the workspace being set as a distribution
+        # if the flood and input workspaces have the same Y units. We need to set the resulting workspace back to
+        # a non-distribution as it may be dimensionless after the divide, but it shouldn't be a distribution.
+        if isinstance(corrected_ws, WorkspaceGroup):
```

Comment:
Is there any situation where we want the output of apply flood workspace as a distribution ? If this is the case we could make this change on the apply flood workspace algorithm instead of here.

Reply:
That's a good thought and something that we might do at some point in the future, but it would need careful testing as that might change results in other parts of the code that use `ApplyFloodWorkspace`. For now I've been discussing the underlying behaviour in the `Divide` algorithm with Richard and Pete to decide whether we should be providing a way of overriding the default behaviour there. If we did that then we could review some of our workflows more generally to make use of that, but it will be a bigger investigation and out of scope of this PR I think.